### DISCUSSION TITLE
how to get selected component properties like Text have default font size and i am creating custom style manager and then i need existing style.

### DISCUSSION THREAD

**Discussion by sohaib389** [GrapesJS/grapesjs#4927] [Ideas]
hi, I am creating custom style manager and I need to set value of existing property, In my case I have create an input type=text for font size of selected component. but very first on selected component i need to show its first font size. in text box.
Thanks 
@artf

**Comment by artf** [comment]
Hi @sohaib389 as you can see from the [customization](https://grapesjs.com/docs/modules/Style-manager.html#customization) section, the value of the custom style manager property is taken from [`Property.getValue`](https://grapesjs.com/docs/api/property.html#getvalue) which updates on component selection (the value itself is based on the current CSS property). 

That means values are based on the current CSS style property of your selection, in case there is no value for the currently selected component of that property, the default one can be displayed ([`Property.getDefaultValue()`](https://grapesjs.com/docs/api/property.html#getdefaultvalue)) and that one can be specified in your [style property definition via `default`](https://grapesjs.com/docs/api/property.html#properties)
